### PR TITLE
feat: Use upload session `urls` for chunk upload

### PR DESCRIPTION
**Pull request by lukaszsocha2**
Closes: SDK-3836

### CODE REVIEW

File: boxsdk/object/upload_session.py [hunk 1]
```diff
@@ -20,10 +21,20 @@ class UploadSession(BaseObject):
...
     _parent_item_type = 'file'
 
-    def get_url(self, *args: Any) -> str:
+    def __init__(
+            self, session: Session, object_id: str, response_object: dict = None, use_upload_session_urls: bool = True
+    ):
+        super().__init__(session, object_id, response_object)
+        self._use_upload_session_urls = use_upload_session_urls
+
+    def get_url(self, *args: Any, url_key: str = None) -> str:
```
Comment (congminh1254) [inline boxsdk/object/upload_session.py:30]: In case of existing customer using custom `API. UPLOAD_URL` they will be use the new URL after this change?

Comment (lukaszsocha2) [reply boxsdk/object/upload_session.py:30]: Yes, unless they set use_upload_session_urls=False

Comment (congminh1254) [reply boxsdk/object/upload_session.py:30]: So in case you are aware of this situation, this PR is good to go :D

**Review by congminh1254 — APPROVED** [review]

**Review by congminh1254 — APPROVED** [review]
LGTM :+1:

**Pull request merged**